src/containers/SessionMenu.js

```diff
@@ -47,17 +66,145 @@ class SessionMenu extends Component {
+            hasCancelButton: false,
+          },
+        }, () => {
+          this.props.openModal('UPDATE_DIALOG');
+        });
+      },
+    ).catch(
+      // Error while checking for updates
+      (error) => {
+        this.setState(...this.state, {
+          updateDialog: {
+            title: 'An Error Occured',
+            type: 'error',
+            content: 'An error has occured while checking for an update. More information to help diagnose the issue can be found below.',
+            additionalInformation: error,
+            onConfirm: () => {},
+            confirmLabel: 'Okay',
+            hasCancelButton: false,
+          },
+        }, () => {
+          this.props.openModal('UPDATE_DIALOG');
+        });
+      },
+    );
+  }
+
+  confirmUpdateDownload() {
+    // User clicked download Button
+    // TODO: implement progress updates/loader
+    // Trigger the download. Returns promise.
+    updater.downloadUpdate().then(
+      // Update downloaded
+      () => {
+        this.setState(...this.state, {
+          updateDialog: {
+            title: 'Update Ready to Install',
+            type: 'warning',
+            additionalInformation: '',
+            content: 'Your update is ready to install. Network Canvas will now close, and will reopen once the update is configured. Make sure you have saved any data within the app before continuing.',
+            onConfirm: () => { updater.installUpdate(); },
+            confirmLabel: 'Install and Restart',
+            hasCancelButton: true,
+          },
+        }, () => {
+          this.props.openModal('UPDATE_DIALOG');
+        });
+      },
+      // Update failed to download. Either cancelled or errored.
+      (failure) => {
+        // Test for error
+        if (failure instanceof Error) {
+          throw new Error(failure.message);
+        }
+        // Otherwise assumed cancelled.
+        this.setState(...this.state, {
+          updateDialog: {
+            title: 'Error Encountered During Update',
+            type: 'error',
+            additionalInformation: '',
+            content: 'An error has occured during the update process.',
+            onConfirm: () => {},
+            confirmLabel: 'Okay',
+            hasCancelButton: false,
+          },
+        }, () => {
+          this.props.openModal('UPDATE_DIALOG');
+        });
+      },
+    ).catch(
+      // Error while downloading update
+      (error) => {
+        this.setState(...this.state, {
+          updateDialog: {
+            title: 'An Error Occured',
+            type: 'error',
+            content: 'An error has occured during the update process. More information to help diagnose the issue can be found below.',
+            additionalInformation: error,
+            onConfirm: () => {},
+            confirmLabel: 'Okay',
+            hasCancelButton: false,
+          },
+        });
+        // This shouldn't be here. But setState is async. How to handle better?
```

Comment:
An `await` or something?

Reply:
It's weird. I tried using the setState callback but that didn't work. Not sure about await, but I'll try it.